lib/datadog/tracing/contrib/action_mailer/events/deliver.rb

```diff
@@ -30,15 +30,20 @@ def span_type
               Tracing::Metadata::Ext::AppTypes::TYPE_WORKER
             end
 
-            def process(span, event, _id, payload)
+            def on_start(span, event, _id, payload)
               super
 
-              span.type = span_type
+              span.set_tag(Tracing::Metadata::Ext::TAG_OPERATION, Ext::TAG_OPERATION_DELIVER)
+            end
+
+            def on_finish(span, event, _id, payload)
+              super
+
+              span.resource = payload[:mailer]
```

Comment:
Is there a reason setting the resource in `on_finish` instead of `on_start` ? Would it because of the data availability?

Reply:
Correct! The data is not available on `on_start`. I'll document that.